sematic/api_client.py

```diff
@@ -97,15 +106,136 @@ def _put(endpoint, json_payload) -> Any:
+    raise error
+
+
+def _validate_server_compatibility_no_catch() -> Tuple[
+    Optional[requests.Response], Optional[Exception]
+]:
+    headers = {"Content-Type": "application/json"}
+    response = None
+    error: Optional[Exception] = None
+    try:
+        response = requests.get(_url("/meta/versions"), headers=headers)
+    except ConnectionError:
+        error = APIConnectionError(
+            (
+                f"Unable to connect to the Sematic API at "
+                f"{get_config().server_url}.\n Make sure the "
+                f"correct server address is set with\n"
+                f"\t$ sematic settings set "
+                f"{SettingsVar.SEMATIC_API_ADDRESS.value} <address>"
+            )
+        )
+        return None, error
+    unexpected_server_response_error = IncompatibleClientError(
+        "The Sematic server did not provide information about its version "
+        "in the expected format. It could be that the server is too old to "
+        "provide this information. Consider upgrading your server if possible, "
+        "or reach out to Sematic for support."
+    )
+    try:
+        response_json = response.json()
+    except JSONDecodeError:
+        return response, unexpected_server_response_error
+    if 400 <= response.status_code < 500:
+        return response, unexpected_server_response_error
+    if response.status_code >= 500:
+        error = ServerError(
+            f"The Sematic server is not responsive, and returned a status code of "
+            f"{response.status_code}."
+        )
+        return response, error
+
+    server_version = cast(Tuple[int, int, int], tuple(response_json["server"]))
+    server_min_client_version = cast(
+        Tuple[int, int, int], tuple(response_json["min_client_supported"])
+    )
+    if CURRENT_VERSION > server_version:
+        error = IncompatibleClientError(
+            f"The Sematic API client is at version "
+            f"{version_as_string(CURRENT_VERSION)}, which is newer than "
+            f"the Sematic server's version of "
+            f"{version_as_string(server_version)}. Please downgrade your client "
+            f"to {version_as_string(server_version)}, or ask your server admin "
+            f"to upgrade to at least {version_as_string(CURRENT_VERSION)}."
+        )
+        return response, error
+    if CURRENT_VERSION < server_min_client_version:
+        error = IncompatibleClientError(
+            f"The Sematic API client is at version "
+            f"{version_as_string(CURRENT_VERSION)}, which is older than the "
+            f"minimum version the server supports. Please upgrade your client "
+            f"to a version of Sematic of at least "
+            f"{version_as_string(server_min_client_version)}"
+        )
+        return response, error
+
+    logger.info(
+        "Sematic client %s is compatible with server %s",
+        version_as_string(CURRENT_VERSION),
+        version_as_string(server_version),
+    )
```

Comment:
Not sure we need to log if things are ok. Or maybe as a `debug`.

Reply:
changed to a debug statement